C7GameData/AIData/TilePath.cs

```diff
@@ -2,12 +2,13 @@
 
 namespace C7GameData
 {
+
 	public class TilePath
 	{
 		private Tile destination; //stored in case we need to re-calculate
 		private Queue<Tile> path;
 
-		private TilePath() {}
+		public TilePath() {}
```

Comment:
Does this need to be public? The project still compiles if it's left as private. I'd rather we don't have parameterless public constructors unless we need them.

Reply:
Yes, in order to deserialize TilePaths, the parameterless constructor needs to be public

Reply:
I see. I'm used to C++ where `private` is only enforced when compiling. It's inconvenient that we're forced to include these constructors since they prevent us from enforcing RAII.

Reply:
Agreed. I can add a comment explaining why it's public at least, but it's unfortunate that it's enforced at runtime